pkg/tls/self_signed_cert.go

```diff
@@ -0,0 +1,205 @@
+	slog.Debug("About to write CA file", "path", caCertPath)
+	writeCAErr := os.WriteFile(caCertPath, caCertPEM, certFilePermissions)
+	if writeCAErr != nil {
+		return &x509.Certificate{}, &ecdsa.PrivateKey{}, fmt.Errorf(
+			"failed to write ca file: %w",
+			writeCAErr,
+		)
+	}
+
+	return caCert, caKeyPair, nil
+}
+
+// nolint: revive
+func GenerateServerCert(hostnames []string, caPath, certPath, keyPath string) (existingCert bool, err error) {
+	// Check for and return existing cert if it already exists
+	existingCert, existingCertErr := ReturnExistingCert(certPath)
+	if existingCertErr != nil {
+		return false, fmt.Errorf("error reading existing certificate data: %w", existingCertErr)
+	}
+	if existingCert {
+		return true, nil
+	}
+
+	// Get the local time zone
+	locationCurrentzone, locErr := time.LoadLocation("Local")
+	if locErr != nil {
+		return false, fmt.Errorf("error detecting local timezone: %w", locErr)
+	}
+	now := time.Now().In(locationCurrentzone)
+
+	// Create CA first
+	caCert, caKeyPair, caErr := GenerateCA(now, caPath)
+	if caErr != nil {
+		return false, fmt.Errorf("error generating certificate authority: %w", caErr)
+	}
+
+	// Generate key pair for the server certficate
+	servKeyPair, servKeyErr := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if servKeyErr != nil {
+		return false, fmt.Errorf("failed to generate server keypair: %w", servKeyErr)
+	}
+
+	servTemplate := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			Organization: []string{caOrganization},
+		},
+		NotBefore:   now.Add(-time.Minute),
+		NotAfter:    now.AddDate(1, 0, 0), // 1 year
+		KeyUsage:    x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature,
+		ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		DNSNames:    hostnames,
+	}
+
+	servRequest := certReq{
+		template:   &servTemplate,
+		parent:     caCert,
+		publicKey:  &servKeyPair.PublicKey,
+		privateKey: caKeyPair,
+	}
+
+	// Generate server certficated signed by the CA
+	_, servCertPEM := genCert(&servRequest)
+	if len(servCertPEM) == 0 {
+		return false, errors.New("error generating server certificate")
+	}
+
+	// Write the certificate to a file
+	writeCertErr := os.WriteFile(certPath, servCertPEM, certFilePermissions)
+	if writeCertErr != nil {
+		return false, fmt.Errorf("failed to write certificate file: %w", writeCertErr)
+	}
+
+	// Write the private key to a file
+	servKeyBytes, marshalErr := x509.MarshalECPrivateKey(servKeyPair)
+	if marshalErr != nil {
+		return false, fmt.Errorf("failed to marshal private key file: %w", marshalErr)
+	}
+	b := pem.Block{Type: "EC PRIVATE KEY", Bytes: servKeyBytes}
+	servKeyPEM := pem.EncodeToMemory(&b)
+	writeKeyErr := os.WriteFile(keyPath, servKeyPEM, keyFilePermissions)
+	if writeKeyErr != nil {
+		return false, fmt.Errorf("failed to write key file: %w", writeKeyErr)
+	}
+
+	return false, nil
+}
+
+func ReturnExistingCert(certPath string) (bool, error) {
```

Comment:
```suggestion
func DoesCertAlreadyExist(certPath string) (bool, error) {
```

Think the function name could be changed to make it more clear what the function is doing